Description: Fix instance boot when Ceph is used for ephemeral storage
 is_block_dev attribute was mistakenly set to True for RBD images (we don't
 actually map RBD images to block devices in the *host* system, but only in
 *guests*, so as far as the host system is concerned an RBD image must not be
 treated like a system block device). This led to a situation when mkfs was
 called for a non existing file and failed when trying to create an ephemeral
 fs.
From 3afdfc5b2cd0af88764a6841690c00e495adf3b9 Mon Sep 17 00:00:00 2001
From: Roman Podoliaka <rpodolyaka@mirantis.com>
Date: Fri, 22 Aug 2014 20:01:33 +0300
Origin: upstream, https://github.com/angdraug/nova/commit/3afdfc5b2cd0af88764a6841690c00e495adf3b9
Bug-Ubuntu: https://launchpad.net/bugs/1362221
Change-Id: I54c0d117a53bb61f278b2e137bd29a595548f5a1
Last-Update: 2014-08-29

diff --git a/nova/virt/libvirt/imagebackend.py b/nova/virt/libvirt/imagebackend.py
index 8c3e28b..a7eb324 100644
--- a/nova/virt/libvirt/imagebackend.py
+++ b/nova/virt/libvirt/imagebackend.py
@@ -516,7 +516,7 @@ def snapshot_extract(self, target, out_format):
 
 class Rbd(Image):
     def __init__(self, instance=None, disk_name=None, path=None, **kwargs):
-        super(Rbd, self).__init__("block", "rbd", is_block_dev=True)
+        super(Rbd, self).__init__("block", "rbd", is_block_dev=False)
         if path:
             try:
                 self.rbd_name = path.split('/')[1]
